### DISCUSSION TITLE
Support for NextJS trailingSlashes enabled?

### DISCUSSION THREAD

**Discussion by antho1jp** [nextauthjs/next-auth#4752] [Ideas]
### Question 💬

When calling getSession() or getting session from component interval we are getting a 308 redirect on the call. I believe this is due to having trailingSlashes enabled because the fetch request directly after the 308 is containing a trailing slash 'http://localhost:3000/api/auth/session/' 

Are there any options or additional setup that I need to configure to reduce the number of calls to just the expected one?
![sessionRedirect](https://user-images.githubusercontent.com/16441160/174811521-fa86c1e7-2e0b-4e19-a561-75e44197a8a5.png)

### Using Below Next Config
`
const withPWA = require('next-pwa')
const runtimeCaching = require('next-pwa/cache')

const withBundleAnalyzer = require('@next/bundle-analyzer')({
	enabled: process.env.ANALYZE === 'true',
})

module.exports = withBundleAnalyzer(
	withPWA({
		reactStrictMode: true,
		trailingSlash: true,
		staticPageGenerationTimeout: 5000,
		dynamicStartUrlRedirect: '/en-us/',
		pwa: {
			dest: 'public',
			register: true,
			skipWaiting: true,
			runtimeCaching,
		},
		poweredByHeader: false,
	}),
)
`

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by goliney** [comment]
I'm also curious about this.

**Comment by ThangHuuVu** [comment]
Interesting, I did not know about this options until now. I suppose we could expose an option to match this feature in Next. It should also be already possible to workaround this on userland by setting the redirect on your server right? 👀

**Reply by antho1jp** [reply]
Excuse my ignorance but what is "userland" in reference to? In the above example the host is Vercel and the server is Vercel's Backend for the front end. 

So the request is from the client NextJS app to the BFF /api/session route setup by the Next-Auth package. 

My assumption is that next-auth is just using the default router rules as defined by the folder structure. 

I actually don't know if the Next config "trailingSlashes" just turns all routes without a slash to a redirect. This could be an issue with NextJS if there isn't a way to expose an api route that includes a slash at the end.

**Reply by antho1jp** [reply]
Actually now that I think of it, Next-Auth could just change the getSession() method to call /api/session/ with the slash. This could be enabled by a config setting.

**Comment by kramer99** [comment]
Following this.

`trailingSlash` option for next-auth would be good to match the option in Next.

**Comment by rodrigo-arias** [comment]
Seeing the same redirect on every page when using the `trailingSlash`. Grateful if anyone has a solution or a workaround to save those extra milliseconds.

**Comment by sokactomas** [comment]
How to solve this problem ? I am using next: 13.2.3 next-auth: 4.15.0

when I call /api/auth/session/ I got response /api/auth/session/
when I call /api/auth/session I got response {}

In my production build I have a lot errors

```
[next-auth][error][CLIENT_FETCH_ERROR] 
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11372:11)',
    name: 'TypeError'
  },
  url: 'https://www.example.com/api/auth/session',
  message: 'fetch failed'
}
```
I need to have set on "trailingSlash: true," in next.config.js
Whats the solution?

**Reply by aakash14goplani** [reply]
@sokactomas - I am facing similar situation in PROD as well as in local env.  Were you able to find fix for this problem?
